src/packageurl/contrib/url2purl.py

```diff
@@ -79,6 +79,24 @@ def purl_from_pattern(type_, pattern, uri):
         return PackageURL(type_, **purl_data)
 
 
+def base_parsing(uri):
+    path = unquote_plus(urlparse(uri).path)
+    segments = [seg for seg in path.split('/') if seg]
+    
+    if len(segments)==1: 
+        return None
+
+    namespace = segments[0]
+    name = segments[1]
+
+    data = {
+        'namespace' : namespace,
+        'name' : name,
+        'segments' : segments
+    }
+    return data
```

Comment:
What do you gain from returning a mapping vs. a plain tuple> Also why not a named tuple or mini object of sorts vs. a mapping?